vm/systemSmartContracts/staking.go

```diff
@@ -116,31 +118,33 @@ func NewStakingSmartContract(
-		marshalizer:                      args.Marshalizer,
-		endOfEpochAccessAddr:             args.EndOfEpochAccessAddr,
-		enableStakingEpoch:               args.EpochConfig.EnableEpochs.StakeEnableEpoch,
-		stakingV2Epoch:                   args.EpochConfig.EnableEpochs.StakingV2EnableEpoch,
-		walletAddressLen:                 len(args.StakingAccessAddr),
-		minNodePrice:                     minStakeValue,
-		correctLastUnjailedEpoch:         args.EpochConfig.EnableEpochs.CorrectLastUnjailedEnableEpoch,
-		validatorToDelegationEnableEpoch: args.EpochConfig.EnableEpochs.ValidatorToDelegationEnableEpoch,
-		correctFirstQueuedEpoch:          args.EpochConfig.EnableEpochs.CorrectFirstQueuedEpoch,
+		eei:                                     args.Eei,
+		unBondPeriod:                            args.StakingSCConfig.UnBondPeriod,
+		stakeAccessAddr:                         args.StakingAccessAddr,
+		jailAccessAddr:                          args.JailAccessAddr,
+		numRoundsWithoutBleed:                   args.StakingSCConfig.NumRoundsWithoutBleed,
+		bleedPercentagePerRound:                 args.StakingSCConfig.BleedPercentagePerRound,
+		maximumPercentageToBleed:                args.StakingSCConfig.MaximumPercentageToBleed,
+		gasCost:                                 args.GasCost,
+		minNumNodes:                             args.MinNumNodes,
+		maxNumNodes:                             args.StakingSCConfig.MaxNumberOfNodesForStake,
+		marshalizer:                             args.Marshalizer,
+		endOfEpochAccessAddr:                    args.EndOfEpochAccessAddr,
+		enableStakingEpoch:                      args.EpochConfig.EnableEpochs.StakeEnableEpoch,
+		stakingV2Epoch:                          args.EpochConfig.EnableEpochs.StakingV2EnableEpoch,
+		walletAddressLen:                        len(args.StakingAccessAddr),
+		minNodePrice:                            minStakeValue,
+		correctLastUnjailedEpoch:                args.EpochConfig.EnableEpochs.CorrectLastUnjailedEnableEpoch,
+		validatorToDelegationEnableEpoch:        args.EpochConfig.EnableEpochs.ValidatorToDelegationEnableEpoch,
+		correctFirstQueuedEpoch:                 args.EpochConfig.EnableEpochs.CorrectFirstQueuedEpoch,
+		correctJailedNotUnstakedEmptyQueueEpoch: args.EpochConfig.EnableEpochs.CorrectJailedNotUnstakedEmptyQueueEpoch,
 	}
 	log.Debug("staking: enable epoch for stake", "epoch", reg.enableStakingEpoch)
 	log.Debug("staking: enable epoch for staking v2", "epoch", reg.stakingV2Epoch)
 	log.Debug("staking: enable epoch for correct last unjailed", "epoch", reg.correctLastUnjailedEpoch)
 	log.Debug("staking: enable epoch for validator to delegation", "epoch", reg.validatorToDelegationEnableEpoch)
 	log.Debug("staking: enable epoch for correct first queued", "epoch", reg.correctFirstQueuedEpoch)
+	log.Debug("staking: enable epoch for correct jailed not unstaked with empty queue", "epoch", reg.flagCorrectJailedNotUnstakedEmptyQueue)
```

Comment:
you need to add log in node/nodeRunner.go as well

Reply:
fixed

Reply:
req.correctJailedNotUnstakedEmptyQueueEpoch instead reg.flagCorrectJailedNotUnstakedEmptyQueue